test/ping/test_json.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+"""
+SPDX-License-Identifier: GPL-2.0-or-later
+Copyright (c) 2024 Georg Pfuetzenreuter <mail+ip@georg-pfuetzenreuter.net>
+"""
+
+from json import loads
+from os import getenv
+from subprocess import Popen, PIPE
+from sys import argv, exit
+
+if getenv('VARIANT') == 'cross-compile':
+  exit(77)
+
+count = 2
+
+if len(argv) > 1:
+  ping = argv[1]
+else:
+  ping = 'builddir/ping/ping'
+
+for arguments in [
+  ['-4', 'localhost'],
+  ['::1'],
+  ['-V'],
+  ['-s5000', '-Mdo', 'opensuse.org'],
+]:
+  for i, line in enumerate(Popen([ping, '-jc', str(count)] + arguments, stdout=PIPE).stdout, 1):
+    print(line)
+    data = loads(line)
+    print(data)
+
+    assert isinstance(data, dict), 'data type does not match'
+
+    if arguments[0] == '-V':
+      assert 'version' in data, f'Missing "version"'
+      continue
```

Comment:
Why to test version in JSON test? But anyway, this will be rewritten anyway (unfortunately to perl, because I haven't find python implementation which would work as easily as perl `Test::Command`).

Reply:
To assess whether the version information was printed in JSON notation as expected.

Reply:
Yeah, unlike errors, `-V` gets into JSON. Sorry for the noise.